presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RewriteAggregationIfToFilter.java

```diff
@@ -118,9 +129,13 @@ public Result apply(AggregationNode aggregationNode, Captures captures, Context
         Assignments.Builder newAssignments = Assignments.builder();
         newAssignments.putAll(sourceProject.getAssignments());
 
-        // Map from the aggregation reference to the IF condition reference.
+        // Map from the aggregation reference to the IF condition reference which will be put in the mask.
         Map<VariableReferenceExpression, VariableReferenceExpression> aggregationReferenceToConditionReference = new HashMap<>();
+        // Map from the aggregation reference to the IF result reference. This only contains the aggregates where the IF can be safely unwrapped.
+        // E.g., SUM(IF(CARDINALITY(array) > 0, array[1])) will not be included in this map as array[1] can return errors if we unwrap the IF.
+        Map<VariableReferenceExpression, VariableReferenceExpression> aggregationReferenceToIfResultReference = new HashMap<>();
 
+        boolean unwrapIfEnabled = getAggregationIfToFilterRewriteStrategy(context.getSession()).ordinal() >= AggregationIfToFilterRewriteStrategy.UNWRAP_IF.ordinal();
```

Comment:
We can just directly compare `== UNWRAP_IF`

Reply:
Done.